web/app.js

```diff
@@ -2046,6 +2046,69 @@ function handleMouseWheel(evt) {
 window.addEventListener('DOMMouseScroll', handleMouseWheel);
 window.addEventListener('mousewheel', handleMouseWheel);
 
+function touchSwipeDetector(evt) {
+  var pdfViewer = PDFViewerApplication.pdfViewer;
+  if (!pdfViewer.isInPresentationMode) {
+    // We only need to handle touch events in presentation mode, when "normal"
+    // scrolling doesn't work.
+    return;
+  }
+
+  // Must move at least these many CSS pixels for it to count as a swipe
+  var SWIPE_MIN_DISTANCE_THRESHOLD = 50;
+  // The swipe angle is allowed to deviate from the x or y axis by this much
+  // before it is not considered a swipe in that direction any more. If this
+  // is set to 45 degrees (Math.PI / 4) then everything will be a valid swipe.
+  var SWIPE_ANGLE_THRESHOLD = Math.PI / 6;
+
+  if (evt.touches.length > 1) {
+    // Multiple touch points detected, cancel the swipe.
+    delete this._startX;
+    delete this._startY;
+    delete this._endX;
+    delete this._endY;
+    return;
+  }
+  switch (evt.type) {
+    case 'touchstart':
+      this._startX = evt.touches[0].pageX;
+      this._startY = evt.touches[0].pageY;
+      break;
+    case 'touchmove':
+      this._endX = evt.touches[0].pageX;
+      this._endY = evt.touches[0].pageY;
+      // Do a preventDefault to avoid the swipe from triggering browser
+      // gestures (Chrome in particular has some sort of swipe gesture in
+      // fullscreen mode).
+      evt.preventDefault();
+      break;
+    case 'touchend':
+      if (typeof this._startX === 'undefined' ||
+          typeof this._endX === 'undefined' ||
+          !PDFViewerApplication.pdfPresentationMode) {
+        return;
+      }
+      var dx = this._endX - this._startX;
+      var dy = this._endY - this._startY;
+      var angle = Math.atan2(dy, dx);
```

Comment:
Nit: Instead of recomputing `Math.abs(angle)` a couple of times below, how about just doing it once and cache the result here instead, e.g.
```javascript
var absAngle = Math.abs(Math.atan2(dy, dx));
```